Tests/PackageGraphTests/ModulesGraphTests.swift

```diff
@@ -2283,7 +2283,7 @@ struct ModulesGraphTests {
         let observability = try loadUnsafeModulesGraph(toolsVersion: .v6_1)
 
         #expect(observability.diagnostics.count == 3)
-        try testDiagnostics(observability.diagnostics) { result in
+        testDiagnostics(observability.diagnostics) { result in
```

Comment:
issue: Since this suite is SwiftTesting, we should be using `expectDiagnostics` instead of `testDiagnostics`

This should probably be handled in a subsequent change